src/mds/CDentry.cc

```diff
@@ -480,22 +480,26 @@ void CDentry::decode_lock_state(int type, const bufferlist& bl)
 }
 
 
-ClientLease *CDentry::add_client_lease(client_t c, Session *session) 
-{
-  ClientLease *l;
-  if (client_lease_map.count(c))
-    l = client_lease_map[c];
-  else {
-    dout(20) << __func__ << " client." << c << " on " << lock << dendl;
-    if (client_lease_map.empty()) {
+client_t ClientLease::get_client() const
+{
+  return session->get_client();
+}
+
+ClientLease *CDentry::add_client_lease(Session *session)
+{
+  client_t client = session->get_client();
+  auto em = client_leases.emplace(std::piecewise_construct,
+				  std::forward_as_tuple(client),
+				  std::forward_as_tuple(this, session));
+  ClientLease *l = &em.first->second;
+  if (em.second) {
```

Comment:
can we use C++ Structured Binding to name the items returned by `client_leases.emplace(...)` rather than using `.first` and `.second`

Reply:
Structured Bindings helps give friendly names to data items and further helps is maintaining continuity when reading and reviewing code instead of flipping back and forth to decipher what `.first` and `.second` is.